src/parser.js

```diff
@@ -2,14 +2,21 @@ import BananaEmitter from './emitter'
 import BananaMessage from './ast'
 
 export default class BananaParser {
-  constructor (locale) {
+  /**
+   *
+   * @param {string} locale
+   * @param {Object} options options
+   * @param {boolean} [options.wikilinks] whether the wiki style link syntax should be parsed or not
+   */
+  constructor (locale, { wikilinks = false } = {}) {
     this.locale = locale
+    this.wikilinks = wikilinks
     this.emitter = new BananaEmitter(this.locale)
   }
 
   parse (message, params) {
-    if (message.includes('{{') || message.includes('[')) {
-      let ast = new BananaMessage(message)
+    if (message.includes('{{') || (this.wikilinks && message.includes('['))) {
+      const ast = BananaMessage(message, { wikilinks: this.wikilinks })
```

Comment:
`new` is not needed here? If so, why?

Reply:
The BananaMessage is a function that returns an ast. Usually a function retuning a non-void value need not require new - what happens is a function call to return a value. Not an instantiation of a class here.
BananaMessage() returns a non-primitive value. The returned ast is an array. So you don't access prototype methods in BananaMessage by ast.x but you access prototype methods of an array.
```
>> function B(){ return [1,2,3];}
undefined
>> B.prototype.m=function(){console.log('mm')}
>> b1=B()
Array(3) [ 1, 2, 3 ]

>> b1.join(',')
"1,2,3"
>>  b2=new B()
Array(3) [ 1, 2, 3 ]

>>  b2.join(',')
"1,2,3"
>> b2.m()
Uncaught TypeError: b2.m is not a function
​```

Reply:
@santhoshtr I think the capitalization is what caused the confusion. The `B` in `BananaMessage` is capitalized - thus implying its a class that has to use `new`.  Or am i missing something?

Reply:
The Abastract Syntax Tree builder in BananaMessage need to be refactored property to an ES6 class. It is not done yet. Then this confusion will go away.

However, that is a candidate for another pullrequest, not related to this.